src/Microsoft.AspNetCore.SystemWebAdapters/Adapters/Features/IHttpApplicationFeature.cs

```diff
@@ -4,13 +4,23 @@
 #if NETCOREAPP
 
 using System;
+using System.Diagnostics.CodeAnalysis;
 using System.Threading;
 using System.Threading.Tasks;
 using System.Web;
 
-namespace Microsoft.AspNetCore.SystemWebAdapters;
+namespace Microsoft.AspNetCore.SystemWebAdapters.Features;
 
-internal interface IHttpApplicationFeature
+/// <summary>
+/// Represents the emulated IIS pipeline and the <see cref="HttpApplication"/> associated with it.
+/// </summary>
+#if NET8_0_OR_GREATER
+[Experimental(Constants.ExperimentalFeatures.DiagnosticId)]
+public
+#else
+internal
+#endif
```

Comment:
This is awkward. We should make these public on all versions and worry about Experimental separately. If there's no convenient Experimental support downlevel then use Obsolete("For testing only, may change in the future") instead.

Reply:
per @joperezr, I've added an internal experimental attribute

Reply:
had a merge issue - they've been removed